src/components/KTreeList/KTreeDraggable.vue

```diff
@@ -0,0 +1,345 @@
+  (event: 'child-change', data: ChildChangeEvent): void,
+  (event: 'selected', item: TreeListItem): void
+}>()
+
+const key = ref(0)
+const internalList = ref<TreeListItem[]>([])
+const draggableAttrs = {
+  tag: 'div',
+  // Note: force-fallback: true is required to override cursor when dragging
+  'force-fallback': true,
+  animation: '100',
+  draggable: '.k-tree-item-container',
+  'item-key': 'id',
+  'ghost-class': 'k-tree-item-dragged',
+  'drag-class': 'k-tree-item-grabbing',
+  class: 'child-drop-zone',
+}
+const dragging = ref(false)
+// using these vars to trick the TS compiler and avoid errors about circular refs in the <template>
+const itemIcon = 'item-icon'
+const itemLabel = 'item-label'
+
+const iconSecondaryColor = (item: TreeListItem): string | undefined => {
+  if (item.icon === 'treeDoc' || !item.icon) {
+    return item.selected ? 'var(--KTreeListItemSelectedBorder, var(--teal-200))' : 'var(--KTreeListItemUnselectedBorder, var(--grey-200))'
+  }
+
+  return undefined
+}
+
+const hasNoChildren = (item: TreeListItem): boolean => {
+  return !internalList.value.filter(anItem => anItem.id === item.id)?.[0].children?.length
+}
+
+const getElementChildren = (item: TreeListItem): TreeListItem[] => {
+  return item.children as TreeListItem[]
+}
+
+const handleChangeEvent = (): void => {
+  if (props.parentId) {
+    emit('child-change', {
+      parent: props.parentId,
+      children: internalList.value,
+    })
+  } else {
+    emit('change', {
+      items: internalList.value,
+    })
+  }
+}
+
+const handleSelectionEvent = (item: TreeListItem) => {
+  emit('selected', item)
+}
+
+const maxLevelReached = computed((): boolean => {
+  return props.level > props.maxLevel
+})
+
+/**
+ * Recursive check to get the maximum depth of an object.
+ * Documented here: https://stackoverflow.com/a/48505969
+ */
+const getMaximumDepth = ({ children = [] }): number => {
+  return children.length === 0 ? 0 : 1 + Math.max(...children.map(getMaximumDepth))
+}
+
+const checkMove = (target: any): boolean => {
+  const levelOfDropLocation = target.relatedContext?.component?.$attrs.level || 0
+  const itemToDrop = target.draggedContext?.element
+
+  // check the level of the deepest nested child in the item being dropped
+  let deepestLevel = 0
+  if (itemToDrop.children?.length) {
+    deepestLevel = getMaximumDepth(itemToDrop)
+  }
+
+  const sumLevel = levelOfDropLocation + deepestLevel
+  if (sumLevel > props.maxLevel) {
+    // disallow movement
+    return false
+  }
+
+  // allow movement
+  return true
+}
+
+const onStartDrag = (): void => {
+  dragging.value = true
+  setDragCursor(true)
+}
+
+const onStopDrag = (): void => {
+  dragging.value = false
+  setDragCursor(false)
+
+  key.value++
+}
+
+// override cursor when dragging
+const setDragCursor = (value: boolean) => {
+  // must be on html element to keep style applied no matter where they drag
+  const html = document.getElementsByTagName('html').item(0)
```

Comment:
Just in case this gets called before mount somehow (unlikely)
```suggestion
  const html = document?.getElementsByTagName('html').item(0)
```